### PR TITLE
standardize taskgraph cache name and location for all models

### LINKED ISSUE #1230: Should we make our `taskgraph` cache directory structure and location consistent across models?
This has become a little more clear due to the `MODEL_SPEC['output']` work just how many different names and directory hierarchies we have for the taskgraph cache directory.  There are varying names and locations across many of the models, and some even write files to the same directory as the cache database.

Now that we're maintaining some additional information in the MODEL_SPEC about the taskgraph cache (and other files, of course), it might be worthwhile to standardize this across all model runs.

### PR DESCRIPTION
**Pull request by emlys**
## Description
Fixes #1230 

I went with `taskgraph_cache` for the name of the directory. I put it at the top level in the workspace because not all models have an intermediate directory.

A few models used a cache directory that contained the `taskgraph.db` file as well as some other less-important intermediate outputs. I split these up so the taskgraph cache has its own directory and the intermediate outputs go in the `intermediate` directory.

Added output specs for SWY cache outputs that were missing them.

## Checklist
- [x] Updated HISTORY.rst and link to any relevant issue (if these changes are user-facing)
- [ ] Updated the user's guide (if needed)
- [ ] Tested the Workbench UI (if relevant)

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -72,6 +72,14 @@ Unreleased Changes
       (`#1327 <https://github.com/natcap/invest/issues/1327>`_)
     * Improved the validation message that is returned when not all spatial
       inputs overlap (`#502 <https://github.com/natcap/invest/issues/502>`_)
+    * Standardized the name and location of the taskgraph cache directory for
+      all models. It is now called ``taskgraph_cache`` and located in the top
+      level of the workspace directory.
+      (`#1230 <https://github.com/natcap/invest/issues/1230>`_)
+* Coastal Blue Carbon
```
Comment (emlys) [inline HISTORY.rst:79]: Not sure why this is showing up in the diff - it was already in main.

File: src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py [hunk 2]
```diff
@@ -1069,18 +1068,17 @@ def _set_up_workspace(args):
...
+        n_workers, reporting_interval=5.0)
 
     suffix = utils.make_suffix_string(args, 'results_suffix')
     intermediate_dir = os.path.join(
         args['workspace_dir'], INTERMEDIATE_DIR_NAME)
     output_dir = os.path.join(
         args['workspace_dir'], OUTPUT_DIR_NAME)
 
-    utils.make_directories([output_dir, intermediate_dir, taskgraph_cache_dir])
+    utils.make_directories([output_dir, intermediate_dir])
```
Comment (dcdenu4) [inline src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py:1081]: TaskGraph creates the taskgraph cache dir if it doesn't exist already, right?

File: src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py [hunk 3]
```diff
@@ -441,18 +493,10 @@
     'flow_dir_mfd_path': 'flow_dir_mfd.tif',
     'qfm_path_list': ['qf_%d.tif' % (x+1) for x in range(N_MONTHS)],
     'stream_path': 'stream.tif',
-}
-
-_TMP_BASE_FILES = {
-    'outflow_direction_path': 'outflow_direction.tif',
-    'outflow_weights_path': 'outflow_weights.tif',
-    'kc_path': 'kc.tif',
```
Comment (dcdenu4) [inline src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py:449]: Were these deletions not being used?

**Review by dcdenu4 — COMMENTED** [review]
Thanks for doing this @emlys , a nice quality of life improvement. Just a few quick questions.

File: src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py (hunk 3 above)
Comment (emlys) [reply src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py:449]: Yes, they were unused

File: src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py (hunk 2 above)
Comment (emlys) [reply src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py:1081]: Correct!

**Review by dcdenu4 — APPROVED** [review]
Thanks Emily!

**Pull request merged**